_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,450 @@
+
+## Other improvements
+
+**Exception histories in the Web UI**
+
+From now on, the Flink Web UI will present up to n last exceptions that caused a job to fail apart
+from the last one.
+
+**Better exception / failure-cause reporting for unsuccessful checkpoints**
+
+Starting from the release Flink will provide statistics for checkpoints which failed or were
+aborted. Prior versions were reporting metrics such as e.g. size of persisted data or trigger time
+only in case a checkpoint succeeded. This change makes it easier to figure out a reason for the
+failure without looking into logs.
+
+**Exactly-once JDBC sink**
+
+From 1.13, JDBC sink can guarantee exactly-once delivery for XA-compliant databases. That means no
+data loss or duplication can occur in case of failover. Double processing (by Flink operators) can
+still occur but the result will only be delivered once.
+
+The feature is available to DataStream programs via the new JdbcSink.exactlyOnceSink method (and
+JdbcXaSinkFunction directly). The target database must be XA-compatible.
+
+**Improved consistency of behavior between different SQL time functions**
+
+**Hive Query syntax compatibility**
+
+In order to improve Hive syntax compatibility, we introduced pluggable `Parser` to the planner
+and added `HiveParser` for Hive dialect. In addition to DDL, Hive dialect now also supports
+commonly-used Hive DML and DQL.
+
+To use Hive dialect, set `table.sql-dialect` to `hive` as you do now, and also make sure to use
+`HiveModule` since some syntax/semantic compatibility features rely on Hive built-in functions.
+Following is an example of using Hive dialect:
+
+```sql
+CREATE CATALOG myhive WITH ('type' = 'hive'); -- setup HiveCatalog
+USE CATALOG myhive;
+LOAD MODULE hive; -- setup HiveModule
+USE MODULES hive,core;
+SET table.sql-dialect = hive; -- enable Hive dialect
+SELECT key, value FROM src CLUSTER BY key; -- run some Hive queries
+```
+Note that Hive dialect no longer supports Flink syntax for DML and DQL. Switch to `default` dialect
+when you need to write Flink syntax.
+
+**Batch support in PyFlink DataStream API**
+
+In 1.12, it has added batch execution mode in the Java DataStream API. In this release, this feature
+is also added in the Python DataStream API.
+
+**General Python User-Defined Aggregate Function Support in group window of PyFlink Table API**
+
+It supported both general Python UDAF and pandas UDAF in PyFlink Table API. In the previous releases,
+The general Python UDAF could only be used in unbounded groupby aggregation. In this release,
+it's also supported to be used in group window of PyFlink Table API.
+
+**Improved performance of Sort-Merge Shuffles**
+
+In 1.12, we introduced sort-merge blocking shuffle to Flink (FLIP-148). In this release the feature
+was optimized to improve usability (fixed direct memory OOM issue) and performance (introduced IO
+scheduling and broadcast optimization).
```

Comment:
Another interesting connector improvement is HBase lookup source supports async lookup and caching now. 

```
**HBase connector supports async lookup and lookup cache**

Before 1.13, HBase lookup source can only be used in sync mode which means the communication between Flink and HBase is synchronous. This leads to poor throughput performance. In 1.12, we introduced async lookup mode for HBase connector and also a lookup cache to reduce I/O interaction. This can make Table/SQL jobs have much better throughput when lookup joining HBase.
```